Workbooks/Workloads/SQL/SQL Insights Onboarding Scripts/Permissions_LoginUser_Account_Creation.ps1

```diff
@@ -0,0 +1,467 @@
+
+                $BSTR_UserId = [System.Runtime.InteropServices.Marshal]::SecureStringToBSTR($AdminUserIdSecret.SecretValue)            
+                $AdminUserId = [System.Runtime.InteropServices.Marshal]::PtrToStringAuto($BSTR_UserId)
+
+                $BSTR_PassKey = [System.Runtime.InteropServices.Marshal]::SecureStringToBSTR($AdminPassKeySecret.SecretValue)            
+                $AdminPassKey = [System.Runtime.InteropServices.Marshal]::PtrToStringAuto($BSTR_PassKey)
+            } catch {
+                # throw exception
+                WriteToConsoleAndLogFile($_)
+                throw 'You have selected SqlAuthWithKeyvault authentication mechanism, however we encountered retrieving credentials from the Keyvault -- '
+            }
+
+            WriteToConsoleAndLogFile("You have selected SqlAuthWithKeyvault authentication mechanism - server credentials found")
+        }
+    } elseif ($SqlAuth) { 
+        if ((([string]::IsNullOrEmpty($AdminUserId)) -or ([string]::IsNullOrEmpty($AdminPassKey)))){
+            # throw exception
+            throw 'You have selected SqlAuth authentication mechanism, this requires AdminUserId and AdminPassKey to be provided'
+        }
+
+        WriteToConsoleAndLogFile("You have selected SqlAuth authentication mechanism - server credentials found")
+    } else {
+        WriteToConsoleAndLogFile("You have not selected an authentication mechanism - defaulting to Sql Auth - server credentials required via prompt")
+        $AdminUserId = Read-Host "Enter server admin user name - "
+        $AdminPassKeySecure = Read-Host -AsSecureString "Enter server admin password - "
+        $BSTR_PassKey = [System.Runtime.InteropServices.Marshal]::SecureStringToBSTR($AdminPassKeySecure)
+        $AdminPassKey = [System.Runtime.InteropServices.Marshal]::PtrToStringAuto($BSTR_PassKey)
+    }
+}
+
+# Name of the login to be created at a server level
+$loginname = 'telegraf'
+
+# Prefix for login password's secret name to be stored in keyvault
+$loginpasswordsecretnameprefix = 'telegraflogin-'
+
+# Name of the user to be created at a database level
+$username = 'telegrafuser'
+
+
+<#
+if (Get-Module -ListAvailable -Name SqlServer) {
+    Write-Host "Module SqlServer exists already"
+} 
+else {
+    Write-Host "Module SqlServer does not exist, installing module"
+    Install-Module -Name SqlServer -Scope CurrentUser
+}
+#>
+
+$SuccessList = New-Object System.Collections.Generic.List[System.Object]
+$FailureList = New-Object System.Collections.Generic.List[System.Object]
+$ConfigConnectionStringList = New-Object System.Collections.Generic.List[System.Object]
+$server_list = @{}
+$success_server_count = 0
+$failure_server_count = 0
+$partial_server_count = 0
+$success_database_count = 0
+$failure_database_count = 0
+[bool] $isDatabaseListDefined = 0
+
+# TODO: Do we want to support this?
```

Comment:
have we resolve this todo?

Reply:
This was an open ended question and I guess it is still open, so for now I can remove this comment, and we can revisit this later when we have a better clarity on it.

Reply:
same with the todo a down below. customers will be running this so don't want them to see bunch of todo's

Reply:
yep makes sense, saw that and took it out too

Reply:
done!